src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -613,6 +615,90 @@ A list of all generated pages. It is an object with one property.
 
 - `pathname` - the finalized path of the page.
 
+### Defining your own hooks for other integrations
+
+You can extend the hooks defined for Astro integration by provide the following type augmentation along with your integration code:
+
+```ts
+// your-integration/types.ts
+declare global {
+  namespace Astro {
+    interface IntegrationHooks {
+      'myLib:eventHappened': (your: string, parameters: number) => Promise<void>;
+    }
+  }
+}
+```
+
+#### Calling custom hooks on other integrations
+
+Custom hooks implemented by all the integrations in use can be accessed from the `astro:config:setup` and `astro:config:done` hooks.
+
+```ts
+// your-integration/index.ts
+import './types.ts';
+
+export default (): AstroIntegration => {
+  return {
+    name: 'your-integration',
+    hooks: {
+      'astro:config:setup': async ({ config }) => {
+        for (const integration of config.integrations) {
+          await integration.hooks['myLib:eventHappened'].?('your values', 123);
+        }
+      },
+    }
+  }
+}
+```
+
+Custom hooks don't need to be triggered inside of those hooks, they can be stored and called later:
+
+```ts
+// your-integration/index.ts
+import './types.ts';
```

Comment:
Realised we kind of skim over this import. Maybe should be mentioned in the introduction that this is needed?

Reply:
It's a bit complicated. It is not required at all; you just have to have some way for the consumer integration to load the type augmentation from the provider integration, and this makes it the most intuitive (by just importing the integration).

But there are multiple ways of doing it and IMHO none of them are intuitive for non-TS-power-users

Reply:
This is in the integration code itself though. If it’s not needed, maybe we remove it?

Reply:
Maybe I wasn't clear. Loading the type augmentation is required, but it isn't required to be done this way.

I'll explain this in the text